include/zephyr/bluetooth/classic/hfp_hf.h

```diff
@@ -53,108 +40,382 @@ struct bt_hfp_hf_cb {
+	 *  If this callback is provided it will be called whenever the
+	 *  unsolicited result code `+BVRA: 1,<vrecstate>` is received from AG.
+	 *  `<vrecstate>`: Bitmask that reflects the current state of the voice
+	 *  recognition engine on the AG.
+	 *  Bit 0 - If it is 1, the AG is ready to accept audio input
+	 *  Bit 1 - If it is 1, the AG is sending audio to the HF
+	 *  Bit 2 - If it is 1, the AG is processing the audio input
+	 *  If @kconfig{CONFIG_BT_HFP_HF_ENH_VOICE_RECG} is not enabled, the
+	 *  unsolicited result code +BVRA will be ignored. And the callback
+	 *  will not be notified.
+	 *
+	 *  @param hf HFP HF object.
+	 *  @param state Value of `<vrecstate>`.
+	 */
+	void (*vre_state)(struct bt_hfp_hf *hf, uint8_t state);
+	/** Textual representation callback
+	 *
+	 *  If this callback is provided it will be called whenever the
+	 *  unsolicited result code `+BVRA: 1,<vrecstate>,
+	 *  <textualRepresentation>` is received from AG.
+	 *  `<textualRepresentation>: <textID>,<textType>,<textOperation>,
+	 *  <string>`.
+	 *  `<textID>`: Unique ID of the current text as a hexadecimal string
+	 *  (a maximum of 4 characters in length, but less than 4 characters
+	 *  in length is valid).
+	 *  `<textType>`: ID of the textType from the following list:
+	 *  0 - Text recognized by the AG from the audio input provided by the HF
+	 *  1 - Text of the audio output from the AG
+	 *  2 - Text of the audio output from the AG that contains a question
+	 *  3 - Text of the audio output from the AG that contains an error
+	 *      description
+	 *  `<textOperation>`: ID of the operation of the text
+	 *  1 - NewText: Indicates that a new text started. Shall be used when the
+	 *               `<textID>` changes
+	 *  2 - Replace: Replace any existing text with the same `<textID>` and
+	 *               same `<textType>`
+	 *  3 - Append: Attach new text to existing text and keep the same
+	 *              `<textID>` and same `<textType>`
+	 *  `<string>`: The `<string>` parameter shall be a UTF-8 text string and
+	 *  shall always be contained within double quotes.
+	 *  If @kconfig{CONFIG_BT_HFP_HF_VOICE_RECG_TEXT} is not enabled, the
+	 *  unsolicited result code +BVRA will be ignored. And the callback
+	 *  will not be notified.
+	 *
+	 *  @param hf HFP HF object.
+	 *  @param id Value of `<textID>`.
+	 *  @param type Value of `<textType>`.
+	 *  @param operation Value of `<textOperation>`.
+	 *  @param text Value of `<string>`.
+	 */
+	void (*textual_representation)(struct bt_hfp_hf *hf, char *id, uint8_t type,
+				       uint8_t operation, char *text);
+	/** Request phone number callback
+	 *
+	 *  If this callback is provided it will be called whenever the
+	 *  result code `+BINP: <Phone number>` is received from AG.
+	 *  If the request is failed, the `number` will be NULL.
+	 *
+	 *  @param hf HFP HF object.
+	 *  @param number Value of `<Phone number>`.
+	 */
+	void (*request_phone_number)(struct bt_hfp_hf *hf, const char *number);
```

Comment:
Please confirm need use struct bt_hfp_hf_call *call as the input parameter for this callback api?

Reply:
It is the callback of function `bt_hfp_hf_request_phone_number`. It is not related to any call. It is used to attach a phone number for a voice tag.